src/Controls/src/Core/CheckBox/CheckBox.cs

```diff
@@ -16,11 +18,38 @@ public partial class CheckBox : View, IElementConfiguration<CheckBox>, IBorderEl
 			BindableProperty.Create(nameof(IsChecked), typeof(bool), typeof(CheckBox), false,
 				propertyChanged: (bindable, oldValue, newValue) =>
 				{
-					((CheckBox)bindable).Handler?.UpdateValue(nameof(ICheckBox.Foreground));
-					((CheckBox)bindable).CheckedChanged?.Invoke(bindable, new CheckedChangedEventArgs((bool)newValue));
-					((CheckBox)bindable).ChangeVisualState();
+					if (bindable is not CheckBox checkBox)
+					{
+						return;
+					}
+
+					checkBox.Handler?.UpdateValue(nameof(ICheckBox.Foreground));
+					checkBox.CheckedChanged?.Invoke(bindable, new CheckedChangedEventArgs((bool)newValue));
+					if (checkBox.Command is not null && checkBox.Command.CanExecute(checkBox.CommandParameter))
+					{
+						checkBox.Command.Execute(checkBox.CommandParameter);
+					}
```

Comment:
Not sure what we should be doing here as we are going to fire the `CheckChanged` event but NOT the command if the command disabled it.

The command can disable the checkbox, and that is fine. But, if the dev then manually sets checked and we only fire one, will that be unexpected? Do we have a place where we have an event and command and we have to do something like this?

I found Button does this:

```cs
public static void ElementClicked(VisualElement visualElement, IButtonElement ButtonElementManager)
{
	if (visualElement.IsEnabled == true)
	{
		ButtonElementManager.Command?.Execute(ButtonElementManager.CommandParameter);
		ButtonElementManager.PropagateUpClicked(); // <- this is the clicked event
	}
}
```